### PR TITLE
chore: Upgrade to `gts@0.5`

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: builder/steps/gen-dockerfile/contents/src/generate_files.ts [hunk 1]
```diff
@@ -77,14 +77,12 @@ export async function generateFiles(
...
-    tool: config.useYarn ? 'yarn' : 'npm',
-    config: config
-  });
+  const dockerfile = renderTemplate(
+      dockerfileTemplate,
+      {baseImage, tool: config.useYarn ? 'yarn' : 'npm', config});
 
   const dockerignoreTemplate = await dataDirReader.read('dockerignore');
-  const dockerignore = renderTemplate(dockerignoreTemplate, {config: config});
+  const dockerignore = renderTemplate(dockerignoreTemplate, {config});
```
Comment (gaofanmichael) [inline builder/steps/gen-dockerfile/contents/src/generate_files.ts]: Is removing `config` expected?

File: builder/steps/gen-dockerfile/contents/test/common.ts [hunk 2]
```diff
@@ -87,6 +87,6 @@ export class MockView implements Reader, Writer, Locator {
   }
 
   async write(path: string, contents: string): Promise<void> {
-    this.pathsWritten.push({path: path, exists: true, contents: contents});
+    this.pathsWritten.push({path, exists: true, contents});
```
Comment (gaofanmichael) [inline builder/steps/gen-dockerfile/contents/test/common.ts]: same here for `path` and `contents`

File: integration/test/test.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ declare type RunCallback =
     (err: Error|null, stdout?: string, stderr?: string) => void;
 
 function run(
-    cmd: string, args: string[], options: {[key: string]: any},
+    cmd: string, args: string[], options: {[key: string]: {}},
```
Comment (gaofanmichael) [inline integration/test/test.ts]: why would we change `any` to an empty object.

File: builder/steps/gen-dockerfile/contents/src/generate_files.ts (hunk 1 above)
Comment (DominicKramer) [reply builder/steps/gen-dockerfile/contents/src/generate_files.ts]: `{config}` is syntactic sugar for `{config: config}` and so it is not unexpected.

File: builder/steps/gen-dockerfile/contents/test/common.ts (hunk 2 above)
Comment (DominicKramer) [reply builder/steps/gen-dockerfile/contents/test/common.ts]: Same as the comment on `{config}`.

File: integration/test/test.ts (hunk 3 above)
Comment (DominicKramer) [reply integration/test/test.ts]: `gts fix` changed this, and I think it is better.  `any` means don't do type checks while `{}` means do type checks but the value can be any object without any constraints on its structure.

Comment (gaofanmichael) [reply integration/test/test.ts]: I guess the benefit is to exclude any primitive types. Then it is better than any.

**Comment by ofrobots** [comment]
> I guess the benefit is to exclude any primitive types. Then it is better than any.

Actually `{}` doesn't guarantee to exclude primitive types. `{}` (same as the `Object` type) specifies that only the properties available on `Object` are available. `any` on the other hand disables all type checks.

To additionally exclude primitives, one can use the `object` type (note the lowercase o).

**Pull request merged**